pkg/server/server_test.go

```diff
@@ -36,3 +37,41 @@ func (s *TestServerSuite) TestServer() {
 	// Check that Engine() returns the router object.
 	require.NotNil(s.T(), srv.Engine())
 }
+
+func (s *TestServerSuite) TestCrossCors() {
+	// We're using the example config for the configuration here as the
+	// specific config params do not matter for testing the routes setup.
+	srv := server.New(s.Config(), fake.NewMockableApplication(s.Config(), nil))
+
+	// Setting up the routes.
+	err := srv.SetupRoutes()
+	require.NoError(s.T(), err)
+
+	// Check that there are routes registered.
+	routes := srv.GetRegisteredRoutes()
+	require.NotEmpty(s.T(), routes)
+
+	// Check that Engine() returns the router object.
+	require.NotNil(s.T(), srv.Engine())
```

Comment:
This is just a copy of `TestServer()`.
Let's add the cors test as a subtest of `TestServer()`:

```
func (s *TestServerSuite) TestServer() {
    ...
    s.T().Run("CORS", func(t *testing.T) {
        // Tests CORS assuming that the server and the routes already setup and tested above
        ...
    })
)

Reply:
fixed here: 5cc1588595e2fc5e68c5173d46d3c495d55db363